test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -6287,6 +6287,26 @@ BOOST_AUTO_TEST_CASE(divisiod_by_zero)
 	BOOST_CHECK(callContractFunction("mod(uint256,uint256)", 7, 0) == encodeArgs());
 }
 
+BOOST_AUTO_TEST_CASE(sdiv_minint_by_negone)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function minInt() returns (int) {
+				return -2**255;
+			}
+			function div(int a, int b) returns (int) {
+				return a / b;
+			}
+			function test() returns (int) {
+				return div(minInt(), -1);
```

Comment:
Are you using multiple methods to avoid the optimizer removing the condition? If so, can you leave a comment?